### ISSUE TITLE
Matmul error introduced in #1884

### ISSUE THREAD

**Issue by hugoabonizio** [huggingface/candle#1905]
After #1884 my inference pipeline stopped working. Until [this commit](https://github.com/huggingface/candle/commit/2a8679509eb55232b37378442c4366343f6dcb11) everything works fine, but after [this one](https://github.com/huggingface/candle/commit/455c42aa729d8019fcb496106478e75dd3246c08) it throws this error: `matmul is only supported for contiguous tensors lstride: [8192, 1] rstride: [1, 4096] mnk: (1, 32000, 4096)`

Reproducing:
```console
$ git checkout 455c42aa729d8019fcb496106478e75dd3246c08
$ cargo run --example llama --release --features "cuda" -- --prompt "hello" --temperature 0.7 --sample-len 200 --dtype f16
loading the model weights from meta-llama/Llama-2-7b-hf
starting the inference loop
Error: matmul is only supported for contiguous tensors lstride: [8192, 1] rstride: [1, 4096] mnk: (1, 32000, 4096)
```

**Comment by LaurentMazare** [comment]
Thanks for reporting this and sorry for the breakage, indeed #1884 was likely to break quite a few things and I haven't tested this model :( this will be fixed in #1906.

**Comment by hugoabonizio** [comment]
@LaurentMazare thanks!
